src/notebooks_service.py

```diff
@@ -147,7 +144,32 @@ def launch_notebook(user, namespace, project, commit_sha, notebook=None):
     if notebook:
         notebook_url += '/notebooks/{notebook}'.format(notebook=notebook)
 
-    return redirect(notebook_url)
+    # 2. redirect to the server when ready
+    if r.status_code == 201:
+        # server is already running
+        return redirect(notebook_url)
+
+    elif r.status_code == 202:
+        # server is spawning - wait a max of 900 seconds = 15 minutes
+        for i in range(450):
```

Comment:
Is there a way to avoid magic numbers?

Reply:
Sure - do you have a suggestion? I'm not sure what an appropriate time to wait would be exactly.

Reply:
What you suggested is fine, it's just that linking 15 minutes to 450*2 happens multiple lines down the comment.
Maybe something like:
```python
timeout = 900 # 900 seconds = 15 minutes
sleep_time = 2
for i in range(timeout / sleep_time):
```

Reply:
You can also use `datetime` objects with a while loop.

Reply:
yep, I intended to parametrize this a bit better, I thought your concern was about the length of the timeout itself.